runtime/src/main/java/org/corfudb/protocols/wireprotocol/ClientHandshakeHandler.java

```diff
@@ -30,7 +30,7 @@
     private final UUID nodeId;
     private final int handshakeTimeout;
     private final HandshakeState handshakeState;
-    private final Queue<CorfuMsg> messages = new ArrayDeque();
+    private final Queue<CorfuMsg> messages = new LinkedList<>();
```

Comment:
I think I'm missing the point, but why is this change needed? A linked list implementation would be better if we remove elements that are not at the tail/head, but I don't see any logic that does that.

Reply:
LinkedList does provide a consistent time (without the amortization of ArrayDeque or without any additional backing array assigned). Also the messages queue is not being randomly accessed, it's purely a FIFO making LinkedList a natural pick. That being said, it's not required per se so if you're against it, I'll keep the ArrayDeque.

Reply:
I was referring to better caching behavior for arrays, but it seems like you introduced a remove (i.e. poll) which didn't exist in the old code. Also, this doesn't seem to be a critical path, so I'm in different.

Reply:
Sounds good. Thanks